File: 2269-count-elements-with-strictly-smaller-and-greater-elements/2269-count-elements-with-strictly-smaller-and-greater-elements.java
class Solution {
    public int countElements(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int min = nums[0];
        int max = nums[0];
        for (int num : nums) {
            if (map.containsKey(num)) {
                map.put(num, map.get(num) + 1);
            } else {
                map.put(num, 1);
            }
            if (num < min) {
                min = num;
            }
            if (num > max) {
                max = num;
            }
        }
        if (min == max) {
            return 0;
        }
        return nums.length - map.get(min) - map.get(max);
    }
}

File: 2269-count-elements-with-strictly-smaller-and-greater-elements/2269-count-elements-with-strictly-smaller-and-greater-elements.py
class Solution:
    def countElements(self, nums: List[int]) -> int:
        counts = {}
        mi = nums[0]
        mx = nums[0]
        for num in nums:
            counts[num] = counts.setdefault(num, 0) + 1
            if num > mx:
                mx = num
            if num < mi:
                mi = num
        if mi == mx:
            return 0
        return len(nums) - counts[mi] - counts[mx] 